passive/google_api_keys_finder.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Google API keys finder by SkypLabs.
+ * https://blog.skyplabs.net
+ * @SkypLabs
+ */
+
+function scan(ps, msg, src) {
+  var alertRisk = 0; // Informational
+  var alertReliability = 3; // High
+  var alertTitle = "Information Disclosure - Google API Keys Found";
+  var alertDesc = "Google API keys have been found.";
+  var alertSolution = "Make sure the API key is not overly permissive.";
+  var cweId = 200; // "Exposure of Sensitive Information to an Unauthorized Actor"
+  var wascId = 13; // "Information Leakage"
+
+  // Regex targeting Google API keys.
+  // Taken from Table III of "How Bad Can It Git? Characterizing Secret Leakagein Public GitHub Repositories".
+  // https://www.ndss-symposium.org/ndss-paper/how-bad-can-it-git-characterizing-secret-leakage-in-public-github-repositories/
+  var re = /AIza[0-9A-Za-z\-_]{35}/g;
+
+  var url = msg.getRequestHeader().getURI().toString();
+
+  // Do not scan unwanted file types.
+  var contentType = msg.getResponseHeader().getHeader("Content-Type");
+  var unwantedFileTypes = ["image/png", "image/jpeg", "image/gif", "application/x-shockwave-flash", "application/pdf"];
+
+  if (unwantedFileTypes.indexOf("" + contentType) >= 0) {
+    return;
+  }
+
+  var body = msg.getResponseBody().toString()
+
+  if (re.test(body)) {
+    re.lastIndex = 0;
+    var foundKeys = [];
+    var comm;
+
+    while (comm = re.exec(body)) {
+      foundKeys.push(comm[0]);
+    }
+
+    ps.raiseAlert(
+      alertRisk,
+      alertReliability,
+      alertTitle,
+      alertDesc,
+      url,
+      "",
+      "",
+      foundKeys.toString(),
+      alertSolution,
+      foundKeys.toString(),
```

Comment:
The evidence shouldn't be everything, but the first match.

Reply:
Good point.

I changed this behaviour in 9167e20. Now, only the first API key is displayed in the "Evidence" field and all the Google API keys are nicely listed in "Other info".